runtime/src/asset.rs

```diff
@@ -1774,4 +1783,30 @@ impl<T: Trait> Module<T> {
 
         Ok(())
     }
+
+    /// Adds a new document
+    fn add_document(ticker: Ticker, doc: Document) {
```

Comment:
Optionally, we can store documents as generic links rather than in special `<Documents<T>>`. That will mean that we don't need to define special add/remove functions for documents. The disadvantage will be that fetching docs might be a little more tedious because there will be other linked data attached to the token. We don't have any other link data attached to the token right now though. That's why I think it's better to just use the existing linkdata infrastructure here.

Reply:
`Asset` module now leverages `add_link` and `remove_link` from `Identity`. Also added `update_link` to `Identity`.